pow/src/lib.rs

```diff
@@ -76,23 +76,22 @@ pub trait MiningWorker {
 
 /// Validates the proof of work of a given header, and that the proof of work
 /// satisfies the requirements of the header.
-pub fn verify_size(bh: &BlockHeader, cuckoo_sz: u32) -> bool {
-	// make sure the pow hash shows a difficulty at least as large as the target
-	// difficulty
-	if bh.difficulty > bh.pow.clone().to_difficulty() {
+pub fn verify_size(bh: &BlockHeader, prev_bh: Option<BlockHeader>, cuckoo_sz: u32) -> bool {
+	// make sure the pow hash shows a difficulty at least as large
+	// as the target difficulty
+	let difficulty = if let Some(prev_bh) = prev_bh {
```

Comment:
That's the tests I would move up to `pipe.rs`, leaving `verify` to only check the validity of the Cucko Cycle.

Reply:
Ok - yeah that makes sense.